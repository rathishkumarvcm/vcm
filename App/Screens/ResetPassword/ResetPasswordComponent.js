import React, { Component } from 'react';
import { Text, View, ScrollView, TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';
import { GIcon, GInputComponent, GHeaderComponent, GFooterComponent } from '../../CommonComponents';
import styles from './styles';
import gblStrings from '../../Constants/GlobalStrings';
import {ValidatePassword} from '../../Utils/ValidatePassword';


class ResetPasswordComponent extends Component {
    constructor(props) {
        super(props);
        this.state = {
            newPassword: '',
            confirmNewPassword: '',
            passwordStrengthStrong: true,
            comparePasswords: true,
        };
        this.disableSubmitButton = true;
    }

    setNewPassword = text => {
        const { newPassword } = this.state;
         this.setState({
             newPassword: text
         });
        this.disableSubmitButton = !((newPassword)===gblStrings.userManagement.strong);
    }

    validateNewPassword = () => {
        const { newPassword } = this.state;
        const validate = (ValidatePassword(newPassword)===gblStrings.userManagement.strong);
        this.setState({ passwordStrengthStrong: validate });
    }

    setConfirmNewPassword = text => {
        this.setState({
            confirmNewPassword: text
        });
    }

    confirmNewlyEnteredPasswords = () => {
        const { confirmNewPassword,newPassword,passwordStrengthStrong } = this.state;
        this.setState({
            comparePasswords: (confirmNewPassword === newPassword)
        });
        if((confirmNewPassword === newPassword)&& (passwordStrengthStrong)){
            this.disableSubmitButton = false;
        }else{
            this.disableSubmitButton = true;
        }
    }

    onClickSubmit = () => {        
        // console.log('Submit Button Clicked...');     
        const { confirmNewPassword } = this.state;  
        const { navigation,saveNewPassword } = this.props;   
        const { navigate } = navigation;  
        const payloadData = {
            newPassword: confirmNewPassword
        };
        saveNewPassword(payloadData);
        navigate('changeLogonCredentials',{message:gblStrings.userManagement.changedPasswordSuccessfully, newPassword:confirmNewPassword});               
    }

    navigateChangeLogonCredentials = () => {
        const { navigation } = this.props;   
        const { navigate } = navigation;  
        navigate('changeLogonCredentials');
    }


    render() { 
        console.log("render--->this.props",JSON.stringify(this.props));
        const { navigation } = this.props;
        const { passwordStrengthStrong,newPassword,comparePasswords,confirmNewPassword } = this.state;
        return (
            <View style={styles.container}>
                <GHeaderComponent navigation={navigation} />
                <ScrollView style={styles.mainFlex}>
                    <TouchableOpacity onPress={this.goBack}>
                        <GIcon
                            name="left"
                            type="antdesign"
                            size={25}
                            color="#707070"
                        />
                    </TouchableOpacity>

                    <Text style={styles.resetYourPasswordText}>{gblStrings.userManagement.restYourPassword}</Text>

                    <View style={styles.line} />

                    <View style={styles.enterNewPasswordFlex}>
                        <Text style={styles.enterPasswordText}>{gblStrings.userManagement.enterNewPassword}</Text>
                        <Text style={styles.explainStyle}>{gblStrings.userManagement.explain}</Text>
                    </View>
                    <GInputComponent
                        propInputStyle={passwordStrengthStrong ? styles.passwordTextBox : styles.passwordTextBoxError}
                        placeholder={gblStrings.userManagement.newPassword}
                        value={newPassword}
                        onChangeText={this.setNewPassword}
                        onBlur={this.validateNewPassword}
                        onSubmitEditing={this.validateNewPassword}
                        validateError={passwordStrengthStrong}
                        secureTextEntry
                    />
                    <View style={styles.passwordStrengthFlex}>
                        <View style={styles.passwordStrongFlex}>
                            <View style={(ValidatePassword(newPassword)===gblStrings.userManagement.strong)?styles.strong:styles.default} />
                            <Text style={styles.strongText}>{gblStrings.userManagement.strong}</Text>
                        </View>
                        <View style={styles.passwordStrongFlex}>
                            <View style={(ValidatePassword(newPassword)===gblStrings.userManagement.good)?styles.good:styles.default} />
                            <Text style={styles.strongText}>{gblStrings.userManagement.good}</Text>
                        </View>
                        <View style={styles.passwordStrongFlex}>
                            <View style={(newPassword.length>0)&&(ValidatePassword(newPassword)===gblStrings.userManagement.weak)?styles.weak:styles.default} />
                            <Text style={styles.strongText}>{gblStrings.userManagement.weak}</Text>
                        </View>
                    </View>

                    <View style={styles.enterPasswordFlex}>
                        <Text style={styles.enterPasswordText}>{gblStrings.userManagement.confirmNewPassword}</Text>
                    </View>

                    <GInputComponent
                        propInputStyle={comparePasswords ? styles.passwordTextBox : styles.passwordTextBoxError}
                        inputStyle={comparePasswords ? null : styles.passwordTextBoxError}
                        placeholder={gblStrings.userManagement.confirmNewPassword}
                        value={confirmNewPassword}
                        onChangeText={this.setConfirmNewPassword}
                        onSubmitEditing={this.confirmNewlyEnteredPasswords}
                        onBlur={this.confirmNewlyEnteredPasswords}
                        errorFlag={!comparePasswords}
                        errorText={gblStrings.userManagement.passwordDoesnotMatch}
                        secureTextEntry
                    />

                    <View style={styles.flex6}>
                        <TouchableOpacity style={styles.backButtonFlex} onPress={this.navigateChangeLogonCredentials}>
                            <Text style={styles.backButtonText}>{gblStrings.common.cancel}</Text>
                        </TouchableOpacity>
                        <TouchableOpacity style={this.disableSubmitButton?styles.submitFlexDisabled:styles.submitFlex} disabled={this.disableSubmitButton} onPress={this.onClickSubmit}>
                            <Text style={styles.submitText}>{gblStrings.common.submit}</Text>
                        </TouchableOpacity>
                    </View>

                    <View style={styles.fullLine} />
                    <View style={styles.tNCFlex}>
                        <Text style={styles.tNcHeader}>{gblStrings.userManagement.VCDiscalimerTitle}{"\n"}</Text>
                        <Text style={styles.tNcBody}>{gblStrings.userManagement.VCDiscalimerDesc}{"\n"}{"\n"}{gblStrings.userManagement.VCPrivacyNoticeDesc} </Text>
                    </View>
                    <GFooterComponent />
                </ScrollView>


            </View>

        );
    }
}


ResetPasswordComponent.propTypes = {
    navigation : PropTypes.instanceOf(Object),
    saveNewPassword : PropTypes.func,
};

ResetPasswordComponent.defaultProps = {
    navigation:{},
    saveNewPassword:()=>{}
};
export default ResetPasswordComponent;