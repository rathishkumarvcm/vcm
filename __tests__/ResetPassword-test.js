/**
  * @jest-environment jsdom
*/

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
import ResetPasswordComponent from '../App/Screens/ResetPassword/ResetPasswordComponent';

describe('Current PIN',()=>{

  //  Snap shot 
  it('snap shot', () => {  
    const tree = renderer.create(<ResetPasswordComponent />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  //  Render using shallow - shallow not includes child component
  test('render',()=>{
    const component = shallow(<ResetPasswordComponent />);  
    component.render();
  });


});