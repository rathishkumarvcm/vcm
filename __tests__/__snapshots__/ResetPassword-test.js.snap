// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current PIN snap shot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F7FAFF",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    clickable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 0.12,
        "flexDirection": "row",
        "opacity": 1,
        "paddingLeft": "4%",
        "paddingRight": "4%",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 100.5,
          "opacity": 1,
          "width": "25%",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../App/Images/logo.png",
          }
        }
        style={
          Object {
            "alignItems": "flex-start",
            "justifyContent": "flex-start",
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "width": "12%",
        }
      }
    />
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginTop": 24.5,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "width": 56,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 40,
              },
              Object {
                "backgroundColor": "transparent",
                "marginHorizontal": 5,
              },
              Object {
                "fontFamily": "EvilIcons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "width": "10%",
        }
      }
    />
    <View
      accessible={true}
      clickable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "height": 30,
            "width": 44,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 30,
              },
              Object {
                "backgroundColor": "transparent",
                "marginHorizontal": 5,
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "height": 30,
            "width": 44,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 30,
              },
              Object {
                "backgroundColor": "transparent",
                "marginHorizontal": 5,
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      clickable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-end",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <RNGestureHandlerButton
        collapsable={false}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          style={
            Object {
              "alignItems": "flex-end",
              "justifyContent": "center",
              "marginEnd": 10,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "height": 40,
                "width": 56,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 40,
                  },
                  Object {
                    "backgroundColor": "transparent",
                    "marginHorizontal": 5,
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              
            </Text>
          </View>
        </View>
      </RNGestureHandlerButton>
    </View>
  </View>
  <RCTScrollView
    style={
      Object {
        "flex": 0.85,
      }
    }
  >
    <View>
      <View
        accessible={true}
        clickable={false}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "height": 25,
              "width": 38,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#707070",
                  "fontSize": 25,
                },
                Object {
                  "backgroundColor": "transparent",
                  "marginHorizontal": 5,
                },
                Object {
                  "fontFamily": "anticon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="iconIcon"
          >
            
          </Text>
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#535353",
            "fontSize": 33,
            "fontWeight": "bold",
            "marginLeft": "4%",
            "marginTop": "4%",
          }
        }
      >
        Reset Password
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#535353",
            "height": 1.5,
            "marginLeft": "4%",
            "marginRight": "4%",
            "marginTop": "4%",
            "opacity": 0.25,
            "width": "92%",
          }
        }
      />
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginTop": 56,
            "paddingLeft": "4%",
            "paddingRight": "4%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#333333DE",
              "fontSize": 26.5,
              "fontWeight": "bold",
              "marginBottom": 13,
            }
          }
        >
          Enter New Password
        </Text>
        <Text
          style={
            Object {
              "color": "#333333DE",
              "fontSize": 26.5,
              "fontWeight": "bold",
              "marginLeft": "3%",
              "textDecorationLine": "underline",
            }
          }
        >
          Explain
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#DEDEDF",
              "borderRadius": 6.5,
              "borderWidth": 1,
              "height": 79,
              "justifyContent": "center",
              "width": "92%",
            },
            Object {
              "color": "#56565A",
              "marginBottom": 29.5,
              "marginLeft": "4%",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          autoFocus={false}
          editable={true}
          inputStyle={Object {}}
          keyboardType="default"
          maxLength={100}
          multiline={false}
          numberOfLines={1}
          onBlur={[Function]}
          onChange={null}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="New Password"
          placeholderTextColor={null}
          propInputStyle={
            Object {
              "color": "#56565A",
              "marginBottom": 29.5,
              "marginLeft": "4%",
            }
          }
          rejectResponderTermination={true}
          returnKeyType="next"
          secureTextEntry={true}
          selectionColor={null}
          style={
            Array [
              Object {
                "alignItems": "center",
                "color": "#B5B5B5",
                "fontSize": 26.5,
                "justifyContent": "center",
                "marginLeft": 23,
              },
              Object {},
            ]
          }
          underlineColorAndroid="transparent"
          validateError={true}
          value=""
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 0.3,
              "flexDirection": "column",
              "marginRight": 10,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#544A54",
                "height": 5,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#56565A",
                "fontSize": 19.5,
                "marginTop": 5,
              }
            }
          >
            Strong
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 0.3,
              "flexDirection": "column",
              "marginRight": 10,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#544A54",
                "height": 5,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#56565A",
                "fontSize": 19.5,
                "marginTop": 5,
              }
            }
          >
            Good
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 0.3,
              "flexDirection": "column",
              "marginRight": 10,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#544A54",
                "height": 5,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#56565A",
                "fontSize": 19.5,
                "marginTop": 5,
              }
            }
          >
            Weak
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 56,
            "paddingLeft": "4%",
            "paddingRight": "4%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#333333DE",
              "fontSize": 26.5,
              "fontWeight": "bold",
              "marginBottom": 13,
            }
          }
        >
          Confirm New Password
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#DEDEDF",
              "borderRadius": 6.5,
              "borderWidth": 1,
              "height": 79,
              "justifyContent": "center",
              "width": "92%",
            },
            Object {
              "color": "#56565A",
              "marginBottom": 29.5,
              "marginLeft": "4%",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          autoFocus={false}
          editable={true}
          errorFlag={false}
          errorText="Both passwords do not match"
          inputStyle={null}
          keyboardType="default"
          maxLength={100}
          multiline={false}
          numberOfLines={1}
          onBlur={[Function]}
          onChange={null}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Confirm New Password"
          placeholderTextColor={null}
          propInputStyle={
            Object {
              "color": "#56565A",
              "marginBottom": 29.5,
              "marginLeft": "4%",
            }
          }
          rejectResponderTermination={true}
          returnKeyType="next"
          secureTextEntry={true}
          selectionColor={null}
          style={
            Array [
              Object {
                "alignItems": "center",
                "color": "#B5B5B5",
                "fontSize": 26.5,
                "justifyContent": "center",
                "marginLeft": 23,
              },
              null,
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "height": 194,
            "marginLeft": "10%",
            "marginRight": "10%",
            "marginTop": "10%",
          }
        }
      >
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": "#61285F45",
              "borderWidth": 1.5,
              "height": 82,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#544A54",
                "fontSize": 26.5,
                "fontWeight": "bold",
              }
            }
          >
            Cancel
          </Text>
        </View>
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#544A54",
              "borderColor": "#61285F45",
              "borderWidth": 1.5,
              "height": 82,
              "justifyContent": "center",
              "marginTop": 29.5,
              "opacity": 0.4,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 26.5,
                "fontWeight": "bold",
              }
            }
          >
            Submit
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#7B8288",
            "height": 1.5,
            "marginTop": 65.5,
            "opacity": 0.4,
          }
        }
      />
      <View
        style={
          Object {
            "marginLeft": "4%",
            "marginTop": 67.5,
            "width": "92%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#56565A",
              "fontSize": 26.5,
              "fontWeight": "bold",
            }
          }
        >
          Investing involves risk including loss of principal.
          

        </Text>
        <Text
          style={
            Object {
              "color": "#56565A",
              "fontSize": 26.5,
            }
          }
        >
          Victory Mutual Funds and USAA Mutual Funds are distributed by Victory Capital Advisers, Inc. (VCA). VictoryShares ETFs and VictoryShares USAA ETFs are distributed by Foreside Fund Services, LLC (Foreside). VCA and Foreside are members of FINRA and SIPC. Victory Capital Management Inc. (VCM) is the investment adviser to the Victory Mutual Funds, USAA Mutual Funds, VictoryShares ETFs and VictoryShares USAA ETFs. VCA and VCM are not affiliated with Foreside. USAA is not affiliated with Foreside, VCM, or VCA. USAA and the USAA logos are registered trademarks and the USAA Mutual Funds and USAA Investments logos are trademarks of United Services Automobile Association and are being used by Victory Capital and its affiliates under license.
          

          

          Victory Capital means Victory Capital Management Inc., the investment manager of the USAA 529 College Savings Plan (Plan). The Plan is distributed by Victory Capital Advisers, Inc., a broker dealer registered with FINRA and an affiliate of Victory Capital. Victory Capital and its affiliates are not affiliated with United Services Automobile Association or its affiliates. USAA and the USAA logo are registered trademarks and the USAA 529 College Savings Plan logo is a trademark of United Services Automobile Association and are being used by Victory Capital and its affiliates under license.
           
        </Text>
      </View>
      <View>
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../App/Images/logo.png",
            }
          }
          style={
            Object {
              "alignItems": "flex-start",
              "height": 164.5,
              "marginLeft": 24.5,
              "marginTop": 49.5,
              "width": 318.5,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#56565A",
              "fontSize": 26.5,
              "marginLeft": 24.5,
              "marginTop": 24.5,
            }
          }
        >
          Connect with Us
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            clickable={false}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../App/Images/twitter.png",
                }
              }
              style={
                Object {
                  "alignItems": "flex-start",
                  "height": 54,
                  "marginLeft": 24.5,
                  "marginTop": 49.5,
                  "width": 65.5,
                }
              }
            />
          </View>
          <View
            accessible={true}
            clickable={false}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../App/Images/linkedin.png",
                }
              }
              style={
                Object {
                  "alignItems": "flex-start",
                  "height": 54,
                  "marginLeft": 49.5,
                  "marginTop": 49.5,
                  "width": 65.5,
                }
              }
            />
          </View>
          <View
            accessible={true}
            clickable={false}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../App/Images/youtube.png",
                }
              }
              style={
                Object {
                  "alignItems": "flex-start",
                  "height": 54,
                  "marginLeft": 49.5,
                  "marginTop": 49.5,
                  "width": 65.5,
                }
              }
            />
          </View>
          <View
            accessible={true}
            clickable={false}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../App/Images/facebook.png",
                }
              }
              style={
                Object {
                  "alignItems": "flex-start",
                  "height": 54,
                  "marginLeft": 49.5,
                  "marginTop": 49.5,
                  "width": 65.5,
                }
              }
            />
          </View>
          <View
            accessible={true}
            clickable={false}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../App/Images/insta.png",
                }
              }
              style={
                Object {
                  "alignItems": "flex-start",
                  "height": 54,
                  "marginLeft": 49.5,
                  "marginTop": 49.5,
                  "width": 65.5,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "flexDirection": "row",
              "height": 82,
              "justifyContent": "center",
              "marginTop": 16.5,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#5D83AE",
                "fontSize": 26.5,
                "fontWeight": "bold",
                "paddingLeft": "4%",
                "width": "50%",
              }
            }
          >
            Privacy Policy
          </Text>
          <Text
            style={
              Object {
                "color": "#5D83AE",
                "fontSize": 26.5,
                "fontWeight": "bold",
                "paddingLeft": "4%",
                "width": "50%",
              }
            }
          >
            Fund Documents
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "flexDirection": "row",
              "height": 82,
              "justifyContent": "center",
              "marginTop": 16.5,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#5D83AE",
                "fontSize": 26.5,
                "fontWeight": "bold",
                "paddingLeft": "4%",
                "width": "50%",
              }
            }
          >
            User Agreements
          </Text>
          <Text
            style={
              Object {
                "color": "#5D83AE",
                "fontSize": 26.5,
                "fontWeight": "bold",
                "paddingLeft": "4%",
                "width": "50%",
              }
            }
          >
            Support
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#56565A",
              "height": 82,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
              }
            }
          >
            Copyright Victory Capital Management Inc. ©2020
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
